src/Workleap.Extensions.Mongo/Indexing/IndexRegistry.cs

```diff
@@ -23,25 +23,30 @@ public IndexRegistry(IEnumerable<Type> documentTypes)
                 throw new InvalidOperationException($"Type '{documentType}' must be decorated with '{nameof(MongoCollectionAttribute)}'");
             }
 
-            var indexProviderType = mongoCollectionAttribute.IndexProviderType ?? typeof(EmptyMongoIndexProvider<>).MakeGenericType(documentType);
+            this.AddDocumentTypeEntry(documentType, mongoCollectionAttribute.IndexProviderType);
+        }
+    }
 
-            if (!HasPublicParameterlessConstructor(indexProviderType))
-            {
-                throw new InvalidOperationException($"Type {indexProviderType}' must have a public parameterless constructor");
-            }
+    internal void AddDocumentTypeEntry(Type documentType, Type? indexProviderType)
```

Comment:
Suggestion for name since it got me confused a little bit and `Entry` since implementation details
- RegisterTypeIndex
- AddDocumentIndex